src/argilla/server/search_engine.py

```diff
@@ -206,8 +207,9 @@ async def search(
         items = [
             SearchResponseItem(record_id=UUID(hit["_id"]), score=hit["_score"]) for hit in response["hits"]["hits"]
         ]
+        total = response["hits"]["total"]["value"]
```

Comment:
Elasticsearch computes the total number of hits with optimization by default (see here:https://www.elastic.co/guide/en/elasticsearch/reference/8.8/search-your-data.html#track-total-hits)

I we want to return always the proper number, we should add the `track_total_hits=True` argument on the search method to compute it.